File: 0796-rotate-string/0796-rotate-string.py
class Solution:
    def rotateString(self, s: str, goal: str) -> bool:
        if len(s) != len(goal):
            return False
        if s==goal: return True

        t = goal[0]
        for i, j in enumerate(s):
            if j == t:
                if goal == s[i:] + s[:i]:
                    return True
        return False

File: 0796-rotate-string/0796-rotate-string.java
class Solution {
    public boolean rotateString(String s, String goal) {
        int l1 = s.length();
        int l2 = goal.length();
        boolean y = (s + s).contains(goal);
        if(l1==l2 && y)
            return true;
        return false;
    }
}